channel/channel.ml

```diff
@@ -25,6 +25,10 @@ module Make(Flow:V1_LWT.FLOW) = struct
   type +'a io = 'a Lwt.t
   type 'a io_stream = 'a Lwt_stream.t
 
+  exception End_of_file (* at least one user understands this exception *)
```

Comment:
Aha, good catch! I think that might be `Pervasives.End_of_file`, so it still wouldn't be caught by defining another `End_of_file` here in Channel (exceptions are not like polymorphic variants -- if you define them twice in two different modules, they are two different exceptions for the purposes of catching them).  I suspect that `End_of_file` leaked from the socket backend (still needs to be confirmed).  This basically reinforces the "exceptions need to disappear from these interfaces" argument, but this fixup en route is just fine.